app/reconciler/readDbIntoCompareStructure.server.ts

```diff
@@ -0,0 +1,40 @@
+import { get, set } from 'lodash';
+import { getReconciliationData } from '~/models/dns-record.server';
+import { buildDomain } from '../utils';
+
+import type { ReconcilerCompareStructure } from './ReconcilerTypes';
+
+/**
+ * This Fn reads the database `Record` table into a `ReconcilerCompareStructure` type object
+ * to be used for later comparison with Route53
+ */
+const readDbIntoCompareStructure = async (): Promise<ReconcilerCompareStructure> => {
+  const MUTATEDcompareStructure: ReconcilerCompareStructure = {};
```

Comment:
Thinking about this a bit more, and I wonder if we should create a `Reconciler` class, with these structures as properties, and the functions operating on the instance data.  The whole thing would be:

```ts
const reconciler = new Reconciler();
// Could also fold this init() call into reconcile()`
await reconciler.init();
await reconciler.reconcile();
```

Now we don't need to deal with mutations across functions, and can modify things in the instance, which isn't going to be odd to future maintainers.

Reply:
This is an interesting idea. We would not eliminate the issue of mutating the object (I like to treat everything as immutable) but at least it would wrap it.

Making the complete Reconciler a class would create too much & complex code in a single class, but we could make the [readRoute53IntoCompareStructure.server.ts](https://github.com/DevelopingSpace/starchart/pull/441/files#diff-305e5602fea0950a79e73510dd6f343bfa48215e1004b1af0ed28a38df45fd8f) into one (that's the only one that's problemmatic). Maybe the DB one too for consistency

Reply:
Agreed, keep the reconciler spread out across files, but combine the compare structures into a instance that we can pass around and modify as needed.